spec/models/git_repository_spec.rb

```diff
@@ -188,5 +188,24 @@
       repo.refresh
       expect(repo.git_tags.collect(&:name)).to match_array(tag_list)
     end
+
+    context do
+      let(:dir_name) { repo.directory_name }
+
+      it "#destroy" do
+        allow(Dir).to receive(:exist?).with(dir_name).and_return(true)
+        expect(FileUtils).to receive(:rm_rf).with(dir_name)
+
+        repo.destroy
+      end
+
+      it "#destroy with failure" do
+        repo_id = repo.id
+        allow(repo).to receive(:delete_repo_dir).and_raise(MiqException::Error, "wham")
```

Comment:
@gmcculloug Yeah, this actually is fine to be an allow, since as @mkanoor said this is just defining the stub behavior

@mkanoor I would probably structure this differently though, so that it's more clear that this is setup. Something like:

```
describe "#destroy" do
  let(:dir_name) { repo.directory_name }

  context "when repo deletion has no errors" do
    it "deletes the repo" do
      # ...
    end
  end

  context "when there is an error during repo deletion" do
    before do
      allow(repo).to receive(:delete_repo_dir).and_raise(MiqException::Error, "wham")
    end

    it "does not delete the repo" do
      # ...
    end
  end
end